pkg/controller/daemon/daemon_controller.go

```diff
@@ -101,6 +105,8 @@ type DaemonSetsController struct {
 	enqueueDaemonSet func(ds *apps.DaemonSet)
 	// A TTLCache of pod creates/deletes each ds expects to see
 	expectations controller.ControllerExpectationsInterface
+	// dsPodIndexer allows looking up pods by DS UID.
```

Comment:
it allows for using any indexes that are currently registered for the informer.

Let's converge with:
https://github.com/kubernetes/kubernetes/pull/130514/files#diff-073530752026c269afe84022a2d452456b7f7edcc2a71c05068ee601e3e14915R117